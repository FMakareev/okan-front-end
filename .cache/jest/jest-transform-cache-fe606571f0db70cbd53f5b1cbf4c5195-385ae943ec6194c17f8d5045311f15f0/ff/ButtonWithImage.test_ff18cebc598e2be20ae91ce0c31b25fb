0f539b0b1d8ab2b60ccd90cf441f3cae
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _react = _interopRequireDefault(require("react"));

var _reactTestRenderer = _interopRequireDefault(require("react-test-renderer"));

var _enzyme = require("enzyme");

var _playMonocolor = _interopRequireDefault(require("../../assets/icons/monocolor/play.monocolor.svg"));

var _StyledThemeProvider = require("../../styles/StyledThemeProvider");

var _ButtonWithImage = require("./ButtonWithImage");

// import 'jest-styled-components'; // этот импорт нужен чтобы в снэпшоты попали css стили которые генерируются при рендере компонента
test('ButtonWithImage: Рендерится без ошибок', function () {
  var output = (0, _enzyme.shallow)(_react["default"].createElement(_StyledThemeProvider.StyledThemeProvider, null, _react["default"].createElement(_ButtonWithImage.ButtonWithImage, {
    children: 'Button',
    variant: 'large',
    size: 'large'
  })));
  expect(output).toMatchSnapshot();
});
test('ButtonWithImage: Рендерится с иконкой справа', function () {
  var output = (0, _enzyme.shallow)(_react["default"].createElement(_StyledThemeProvider.StyledThemeProvider, null, _react["default"].createElement(_ButtonWithImage.ButtonWithImage, {
    rightIcon: _react["default"].createElement(_playMonocolor["default"], null),
    children: 'Button',
    variant: 'large',
    size: 'large'
  })));
  expect(output).toMatchSnapshot();
});
test('ButtonWithImage: Рендерится с иконкой слева', function () {
  var output = (0, _enzyme.shallow)(_react["default"].createElement(_StyledThemeProvider.StyledThemeProvider, null, _react["default"].createElement(_ButtonWithImage.ButtonWithImage, {
    leftIcon: _react["default"].createElement(_playMonocolor["default"], null),
    children: 'Button',
    variant: 'large',
    size: 'large'
  })));
  expect(output).toMatchSnapshot();
});
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIkJ1dHRvbldpdGhJbWFnZS50ZXN0LmpzIl0sIm5hbWVzIjpbInRlc3QiLCJvdXRwdXQiLCJleHBlY3QiLCJ0b01hdGNoU25hcHNob3QiXSwibWFwcGluZ3MiOiI7Ozs7QUFBQTs7QUFDQTs7QUFDQTs7QUFHQTs7QUFFQTs7QUFDQTs7QUFKQTtBQU1BQSxJQUFJLENBQUMsd0NBQUQsRUFBMkMsWUFBTTtBQUNuRCxNQUFNQyxNQUFNLEdBQUcscUJBQ2IsZ0NBQUMsd0NBQUQsUUFDRSxnQ0FBQyxnQ0FBRDtBQUFpQixJQUFBLFFBQVEsRUFBRSxRQUEzQjtBQUFxQyxJQUFBLE9BQU8sRUFBRSxPQUE5QztBQUF1RCxJQUFBLElBQUksRUFBRTtBQUE3RCxJQURGLENBRGEsQ0FBZjtBQUtBQyxFQUFBQSxNQUFNLENBQUNELE1BQUQsQ0FBTixDQUFlRSxlQUFmO0FBQ0QsQ0FQRyxDQUFKO0FBU0FILElBQUksQ0FBQyw4Q0FBRCxFQUFpRCxZQUFNO0FBQ3pELE1BQU1DLE1BQU0sR0FBRyxxQkFDYixnQ0FBQyx3Q0FBRCxRQUNFLGdDQUFDLGdDQUFEO0FBQWlCLElBQUEsU0FBUyxFQUFFLGdDQUFDLHlCQUFELE9BQTVCO0FBQXNDLElBQUEsUUFBUSxFQUFFLFFBQWhEO0FBQTBELElBQUEsT0FBTyxFQUFFLE9BQW5FO0FBQTRFLElBQUEsSUFBSSxFQUFFO0FBQWxGLElBREYsQ0FEYSxDQUFmO0FBS0FDLEVBQUFBLE1BQU0sQ0FBQ0QsTUFBRCxDQUFOLENBQWVFLGVBQWY7QUFDRCxDQVBHLENBQUo7QUFTQUgsSUFBSSxDQUFDLDZDQUFELEVBQWdELFlBQU07QUFDeEQsTUFBTUMsTUFBTSxHQUFHLHFCQUNiLGdDQUFDLHdDQUFELFFBQ0UsZ0NBQUMsZ0NBQUQ7QUFBaUIsSUFBQSxRQUFRLEVBQUUsZ0NBQUMseUJBQUQsT0FBM0I7QUFBcUMsSUFBQSxRQUFRLEVBQUUsUUFBL0M7QUFBeUQsSUFBQSxPQUFPLEVBQUUsT0FBbEU7QUFBMkUsSUFBQSxJQUFJLEVBQUU7QUFBakYsSUFERixDQURhLENBQWY7QUFLQUMsRUFBQUEsTUFBTSxDQUFDRCxNQUFELENBQU4sQ0FBZUUsZUFBZjtBQUNELENBUEcsQ0FBSiIsInNvdXJjZXNDb250ZW50IjpbImltcG9ydCBSZWFjdCBmcm9tICdyZWFjdCc7XHJcbmltcG9ydCByZW5kZXJlciBmcm9tICdyZWFjdC10ZXN0LXJlbmRlcmVyJztcclxuaW1wb3J0IHsgc2hhbGxvdyB9IGZyb20gJ2VuenltZSc7XHJcblxyXG4vLyBpbXBvcnQgJ2plc3Qtc3R5bGVkLWNvbXBvbmVudHMnOyAvLyDRjdGC0L7RgiDQuNC80L/QvtGA0YIg0L3Rg9C20LXQvSDRh9GC0L7QsdGLINCyINGB0L3RjdC/0YjQvtGC0Ysg0L/QvtC/0LDQu9C4IGNzcyDRgdGC0LjQu9C4INC60L7RgtC+0YDRi9C1INCz0LXQvdC10YDQuNGA0YPRjtGC0YHRjyDQv9GA0Lgg0YDQtdC90LTQtdGA0LUg0LrQvtC80L/QvtC90LXQvdGC0LBcclxuaW1wb3J0IFBsYXkgZnJvbSAnLi4vLi4vYXNzZXRzL2ljb25zL21vbm9jb2xvci9wbGF5Lm1vbm9jb2xvci5zdmcnO1xyXG5cclxuaW1wb3J0IHsgU3R5bGVkVGhlbWVQcm92aWRlciB9IGZyb20gJy4uLy4uL3N0eWxlcy9TdHlsZWRUaGVtZVByb3ZpZGVyJztcclxuaW1wb3J0IHsgQnV0dG9uV2l0aEltYWdlIH0gZnJvbSAnLi9CdXR0b25XaXRoSW1hZ2UnO1xyXG5cclxudGVzdCgnQnV0dG9uV2l0aEltYWdlOiDQoNC10L3QtNC10YDQuNGC0YHRjyDQsdC10Lcg0L7RiNC40LHQvtC6JywgKCkgPT4ge1xyXG4gIGNvbnN0IG91dHB1dCA9IHNoYWxsb3coXHJcbiAgICA8U3R5bGVkVGhlbWVQcm92aWRlcj5cclxuICAgICAgPEJ1dHRvbldpdGhJbWFnZSBjaGlsZHJlbj17J0J1dHRvbid9IHZhcmlhbnQ9eydsYXJnZSd9IHNpemU9eydsYXJnZSd9IC8+XHJcbiAgICA8L1N0eWxlZFRoZW1lUHJvdmlkZXI+LFxyXG4gICk7XHJcbiAgZXhwZWN0KG91dHB1dCkudG9NYXRjaFNuYXBzaG90KCk7XHJcbn0pO1xyXG5cclxudGVzdCgnQnV0dG9uV2l0aEltYWdlOiDQoNC10L3QtNC10YDQuNGC0YHRjyDRgSDQuNC60L7QvdC60L7QuSDRgdC/0YDQsNCy0LAnLCAoKSA9PiB7XHJcbiAgY29uc3Qgb3V0cHV0ID0gc2hhbGxvdyhcclxuICAgIDxTdHlsZWRUaGVtZVByb3ZpZGVyPlxyXG4gICAgICA8QnV0dG9uV2l0aEltYWdlIHJpZ2h0SWNvbj17PFBsYXkgLz59IGNoaWxkcmVuPXsnQnV0dG9uJ30gdmFyaWFudD17J2xhcmdlJ30gc2l6ZT17J2xhcmdlJ30gLz5cclxuICAgIDwvU3R5bGVkVGhlbWVQcm92aWRlcj4sXHJcbiAgKTtcclxuICBleHBlY3Qob3V0cHV0KS50b01hdGNoU25hcHNob3QoKTtcclxufSk7XHJcblxyXG50ZXN0KCdCdXR0b25XaXRoSW1hZ2U6INCg0LXQvdC00LXRgNC40YLRgdGPINGBINC40LrQvtC90LrQvtC5INGB0LvQtdCy0LAnLCAoKSA9PiB7XHJcbiAgY29uc3Qgb3V0cHV0ID0gc2hhbGxvdyhcclxuICAgIDxTdHlsZWRUaGVtZVByb3ZpZGVyPlxyXG4gICAgICA8QnV0dG9uV2l0aEltYWdlIGxlZnRJY29uPXs8UGxheSAvPn0gY2hpbGRyZW49eydCdXR0b24nfSB2YXJpYW50PXsnbGFyZ2UnfSBzaXplPXsnbGFyZ2UnfSAvPlxyXG4gICAgPC9TdHlsZWRUaGVtZVByb3ZpZGVyPixcclxuICApO1xyXG4gIGV4cGVjdChvdXRwdXQpLnRvTWF0Y2hTbmFwc2hvdCgpO1xyXG59KTtcclxuIl19