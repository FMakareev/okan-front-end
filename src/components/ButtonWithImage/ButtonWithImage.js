import React, {Component} from 'react';
import PropTypes from 'prop-types';

/** View */
import Flex from '../Flex/Flex';
import ButtonBase from '../ButtonBase/ButtonBase';
import {PreloaderWrapper, SpeedingWheel} from "@lib/ui/SmallPreloader/SmallPreloader";

export class ButtonWithImage extends Component {
  static propTypes = {
    children: PropTypes.any,
    disabled: PropTypes.any,
    isLoading: PropTypes.bool,
    leftIcon: PropTypes.any,
    ml: PropTypes.number,
    mr: PropTypes.number,
    name: PropTypes.string,
    rightIcon: PropTypes.any,
    size: PropTypes.string,
    variant: PropTypes.string,
    width: PropTypes.any,
    widthIcon: PropTypes.any
  }

  shouldComponentUpdate(nextProps){
    if(nextProps.isLoading !== this.props.isLoading){
      return true;
    }
    if(nextProps.disabled !== this.props.disabled){
      return true;
    }
    return false;
  }

  render() {
    const {
      leftIcon,
      rightIcon,
      children,
      variant,
      size,
      mr,
      ml,
      width,
      widthIcon,
      isLoading,
      ...props
    } = this.props;
    return (
      <ButtonBase variant={variant} size={size} width={width} {...props}>
        <Flex justifyContent={'center'} alignItems={'space-around'} width={'100%'}>
          {leftIcon && (
            <Flex position={'relative'} mr={mr} justifyContent={'center'} alignItems={'center'} width={widthIcon}>
              {leftIcon}
            </Flex>
          )}

          <Flex justifyContent={'center'} alignItems={'center'}>
            {children}
            {
              isLoading &&
              <PreloaderWrapper>
                <SpeedingWheel/>
              </PreloaderWrapper>
            }
          </Flex>

          {rightIcon && (
            <Flex position={'relative'} ml={ml} justifyContent={'center'} alignItems={'center'} width={widthIcon}>
              {rightIcon}
            </Flex>
          )}
        </Flex>
      </ButtonBase>
    );
  }
}


export default ButtonWithImage;
