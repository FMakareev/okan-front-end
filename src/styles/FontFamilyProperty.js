/**
 * @params {string} fontFamily - string name of the font name property in the theme
 * @params {object} theme
 * */
export const FontFamilyProperty = ({ fontFamily, theme }) => {
  try {
    if (fontFamily && theme.fontFamily) {
      return `font-family: '${theme.fontFamily[fontFamily]}';`;
    }
  } catch (e) {
    console.error(e);
  }
  return undefined;
};
export default FontFamilyProperty;
