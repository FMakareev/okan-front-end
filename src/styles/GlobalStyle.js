import { createGlobalStyle } from 'styled-components'


export const GlobalStyle = createGlobalStyle`
  *{
    box-sizing: border-box;
  }
  body, html, #app {
    padding: 0;
    margin: 0;
    width: 100%;
    min-height: 100vh;
    background-color: #fff;
  }
`;
