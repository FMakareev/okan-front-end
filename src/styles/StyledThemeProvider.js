import React from 'react';
import PropTypes from 'prop-types';
import { Provider } from 'rebass';

/** Variants */
import { ButtonVariant } from './variants/buttonVariant';
import { ButtonSize } from './variants/buttonSize';
import { ButtonComment } from './variants/buttonComment';
import { TextVariant } from './variants/textVariant';
import { InputSize } from '@lib/styles/variants/InputSize';
import { InputVariant } from '@lib/styles/variants/InputVariant';

export const ColorPallet = {
  color0: '#FFFFFF',
  color1: '#E5E5E5',
  color2: '#DEDEDE',
  color3: '#4F4F4F',
  color4: '#848484',
  color5: '#007FAF',
  color6: '#006699',
  color7: '#00649C',
  color8: '#007CC4',
  color9: '#F97E63',
  color10: '#AEAEAE',
  color11: '#333333',
  color12: '#DF4624',
  color13: '#f2f2f2',
  color14: '#fff3c1',
};

const Space = [
  0,
  2,
  4,
  8,
  12,
  16,
  18,
  20,
  24,
  28,
  32,
  36,
  40,
  44,
  48,
  52,
  56,
  60,
  64,
  68,
  72,
  76,
  80,
];

const Weight = [300, 500, 700];

const Breakpoints = ['1024px', '1200px', '1440px'];

const boxShadow = [
  '2px 4px 8px 0px rgba(138,138,138,0.5)',
  '2px 4px 8px 0px rgba(138,138,138,0.5)',
  '2px 2px 4px 0px rgba(0,127,175,1)',
];

const textAlign = ['left', 'center', 'right'];

export const StyledThemeProvider = ({ children }) => (
  <Provider
    theme={{
      space: Space,
      fontSizes: Space,
      fontWeight: Weight,
      lineHeight: Space,
      boxShadow,
      borderRadius: Space,
      borderColor: ColorPallet,
      breakpoints: Breakpoints,
      colors: ColorPallet,
      variant: {
        buttons: ButtonVariant,
        buttonSize: ButtonSize,
        buttonComment: ButtonComment,
        text: TextVariant,
        inputSize: InputSize,
        inputVariant: InputVariant,
      },
      fontFamily: {
        primary300: 'Museo Sans 300',
        primary500: 'Museo Sans 500',
        primary700: 'Museo Sans 700',
        secondary: 'Circe Regular',
        secondaryBold: 'Circe Bold',
      },
      textAlign: textAlign,
    }}>
    {children}
  </Provider>
);

StyledThemeProvider.propTypes = {
  children: PropTypes.element.isRequired,
};

StyledThemeProvider.defaultProps = {
  children: 'children is empty',
};

export default StyledThemeProvider;
