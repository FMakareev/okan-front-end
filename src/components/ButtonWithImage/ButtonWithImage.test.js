import React from 'react';
import renderer from 'react-test-renderer';
// import 'jest-styled-components'; // этот импорт нужен чтобы в снэпшоты попали css стили которые генерируются при рендере компонента
import Play from '../../assets/icons/monocolor/play.monocolor.svg';

import { StyledThemeProvider } from '../../styles/StyledThemeProvider';
import { ButtonWithImage } from './ButtonWithImage';

test('ButtonWithImage: Рендерится без ошибок', () => {
  const output = renderer
    .create(
      <StyledThemeProvider>
        <ButtonWithImage children={'Button'} variant={'large'} size={'large'} />
      </StyledThemeProvider>,
    )
    .toJSON();
  expect(output).toMatchSnapshot();
});

test('ButtonWithImage: Рендерится с иконкой справа', () => {
  const output = renderer.create(
    <StyledThemeProvider>
      <ButtonWithImage rightIcon={<Play />} children={'Button'} variant={'large'} size={'large'} />
    </StyledThemeProvider>,
  );
  expect(output).toMatchSnapshot();
});

test('ButtonWithImage: Рендерится с иконкой слева', () => {
  const output = renderer.create(
    <StyledThemeProvider>
      <ButtonWithImage leftIcon={<Play />} children={'Button'} variant={'large'} size={'large'} />
    </StyledThemeProvider>,
  );
  expect(output).toMatchSnapshot();
});
