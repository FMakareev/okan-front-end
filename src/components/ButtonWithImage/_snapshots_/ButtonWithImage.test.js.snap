import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import ButtonWithImage from '../ButtonWithImage';
import Play from '../../../assets/icons/monocolor/play.monocolor.svg';

describe('Button With Image', () => {
  it('Должно правильно отрендерить Button с картинкой справа', () => {
    const output = shallow(
      <ButtonWithImage iconRight={<Play/>} name={'Button'} variant={'large'} size={'large'}/>
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
})

describe('Button With Image', () => {
  it('Должно правильно отрендерить Button с картинкой слева', () => {
    const output = shallow(
      <ButtonWithImage iconleft={<Play/>} name={'Button'} variant={'small'} size={'medium'}/>
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
})
